src/components/SingleDatePicker.jsx

```diff
@@ -248,7 +248,12 @@ class SingleDatePicker extends React.PureComponent {
 
   onFocusOut(e) {
     const { onFocusChange } = this.props;
-    if (this.container.contains(e.relatedTarget || e.target)) return;
+    if (
+      this.dayPickerContainer
+      && this.dayPickerContainer.contains(e.relatedTarget || e.target)
```

Comment:
I’m not sure this is the right change - is it possible the relevant tests were mocking this.container but need to be updated to mock this.dayPickerContainer?

Reply:
Even I thought of that. The fact is that there are always 2 refs - the outer ref `this.container`, and the inner ref `this.dayPickerContainer`. The issue was caused because the wrong container ref was being searched in case of elements rendered in portal. So I referred to `DateRangePicker.jsx`, where I saw that `this.dayPickerContainer` was the one being used in `focusOut` handler. And thus the change.
So should we update the tests, or is the change wrong?

Reply:
Your change might be perfect :-) but maybe you could add tests that would help ensure it's not broken in the future?

Reply:
I've changed the ref being used inside the test, and reverted the previous check. @ljharb